### DISCUSSION TITLE
Crop close to the product size 1600x1600

### DISCUSSION THREAD

**Discussion by TNNSLAB** [ImageMagick/ImageMagick#2146] [Help] [answered]
Hi guys, i have a lot of images that I have to crop in photoshop...thinking to automate somehow this process...have no idea if it would be possible to achieve this. 

I need the files to be cropped to 1600x1600px

![test2](https://user-images.githubusercontent.com/46636486/84655706-b8ebfe00-af11-11ea-8b4e-9432bfd45c3f.jpg)
![586A7666](https://user-images.githubusercontent.com/46636486/84655767-d5883600-af11-11ea-9679-1ee4272a9d8c.jpg)

Any help is appreciated. 

Using Imagemagick on mac

**Comment by snibgo** [comment]
What version of IM?

How about:
```
magick in.tiff -gravity center -crop 1600x1600+0+0 +repage out.tiff
```

To do many files, either put that in a loop, or:
```
magick mogrify -gravity center -crop 1600x1600+0+0 +repage *.tiff
```

**Reply by TNNSLAB** [reply]
Hi @snibgo ! 

Version: ImageMagick 7.0.8-59 Q16 x86_64 2019-08-05 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC

`magick test2.jpg -gravity center -crop 1600x1600+0+0 +repage out.jpg` 

gives me 

![out](https://user-images.githubusercontent.com/46636486/84659430-af659480-af17-11ea-9e9c-20f71887974f.jpg)

I would need something like this : 

![test24](https://user-images.githubusercontent.com/46636486/84659507-cad09f80-af17-11ea-8d56-fef39de265fe.jpg)

**Comment by TNNSLAB** [comment]
Anyone ? any ideas ?

**Comment by snibgo** [comment, marked as answer]
That isn't just a crop, but also a resize.

With this plain background, we can easily find the bounding box of the product:

```
magick 84655706-b8ebfe00-af11-11ea-8b4e-9432bfd45c3f.jpg -fuzz 30% -format %@ info:

1217x2730+2788+981
```

With some arithmetic, you can find the square that encloses that rectangle. Crop to that square, then resize to 1600x1600.
